qiskit/transpiler/passes/utils/remove_labeled_ops.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Remove all babeled ops from a circuit"""
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.utils import control_flow
+
+
+class RemoveLabeledOps(TransformationPass):
+    """Remove all operations with a specific label..
+
+    This transformation pass is used to remove
+
+    Example:
+
+        .. plot::
+           :include-source:
+
+            from qiskit import QuantumCircuit
+            from qiskit.transpiler.passes import RemoveBarriers
+
+            circuit = QuantumCircuit(1)
+            circuit.x(0, label='foo')
+            circuit.barrier()
+            circuit.h(0)
+
+            circuit = RemoveLabeledOps('foo')(circuit)
+            circuit.draw('mpl')
+
+    """
```

Comment:
Forward thinking: would it be cleaner to make this pass `FilterNodes` or something, and have it take a callable `DAGOpNode -> bool`?  The obvious worry with that is pickling the pass managers - I can't remember if #10291 affects that.

Reply:
Yeah, it's probably a better call to make it more general I was too concentrated on solving the issue at hand. It also pairs well with what @danielleodigie added to rustworkx recently with: https://github.com/Qiskit/rustworkx/pull/886. But yeah, the filter functions would be constrained on being pickleable (it uses `dill` by default so that expands things a bit) otherwise the pass manager won't be able to run on multiple processes.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10323/commits/691d7d5e99c587f99482cf9c67e22cd42fee4e91